application/src/main/java/org/togetherjava/tjbot/features/mediaonly/MediaOnlyChannelListener.java

```diff
@@ -54,20 +62,37 @@ private boolean messageHasNoMediaAttached(Message message) {
                 && !message.getContentRaw().contains("http");
     }
 
-    private RestAction<Message> dmUser(Message message) {
+    private MessageEmbed originalMessageEmbed(Message message) {
         String originalMessageContent = message.getContentRaw();
-        MessageEmbed originalMessageEmbed =
-                new EmbedBuilder().setDescription(originalMessageContent)
-                    .setColor(Color.ORANGE)
-                    .build();
+        return new EmbedBuilder().setDescription(originalMessageContent)
+                .setColor(Color.ORANGE)
+                .build();
+    }
+
+    private RestAction<Message> warnUser(Message message) {
+        MessageEmbed originalMessageEmbed = originalMessageEmbed(message);
+
+        long authorId = message.getAuthor().getIdLong();
+
+        MessageCreateData pingMessage = new MessageCreateBuilder().setContent("Hey there, you <@"
+                        + authorId
+                        + "> posted a message without media (image, video, link) in a media-only channel. Please see the description of the channel for details and then repost with media attached, thanks 😀")
+                .setEmbeds(originalMessageEmbed)
+                .build();
+
+        return message.getChannel().sendMessage(pingMessage);
+    }
+
+    private RestAction<Message> dmUser(Message message) {
+        MessageEmbed originalMessageEmbed = originalMessageEmbed(message);
 
         MessageCreateData dmMessage = new MessageCreateBuilder().setContent(
-                "Hey there, you posted a message without media (image, video, link) in a media-only channel. Please see the description of the channel for details and then repost with media attached, thanks 😀")
-            .setEmbeds(originalMessageEmbed)
-            .build();
+                        "Hey there, you posted a message without media (image, video, link) in a media-only channel. Please see the description of the channel for details and then repost with media attached, thanks 😀")
+                .setEmbeds(originalMessageEmbed)
+                .build();
 
         return message.getAuthor()
-            .openPrivateChannel()
-            .flatMap(channel -> channel.sendMessage(dmMessage));
+                .openPrivateChannel()
+                .flatMap(channel -> channel.sendMessage(dmMessage));
```

Comment:
instead have method which gives u MessageCreateBuilder(will the all data, i.e. message and embeds set already) of the common part then send it where u need to for example
```java
message.delete()
    .flatMap(any -> message.getAuthor()
        .openPrivateChannel()
        .flatMap(channel -> channel.sendMessage(getWarnMessage(message).build())))
    .queue( ... );
```

Reply:
Its Exactly the same thing just out of the Function creating 2 more functions. 
does it improve its Efficiency?

Reply:
Ok I will update it